astropy/units/format/latex.py

```diff
@@ -68,8 +68,7 @@ def to_string(
         unit: UnitBase,
         fraction: bool | Literal["inline", "multiline"] = "multiline",
     ) -> str:
-        s = super().to_string(unit, fraction=fraction)
-        return rf"$\mathrm{{{s}}}$"
+        return rf"$\mathrm{{{cls._to_string(unit, fraction=fraction)}}}$"
```

Comment:
This change is not needed for type checking, but it's similar enough to the change made in `Console` that they can be made together in the same commit.

Reply:
I don't really like it - see above.